src/library.js

```diff
@@ -910,13 +910,15 @@ LibraryManager.library = {
     {{{ makeStructuralReturn(['ret', '0']) }}};
   },
 
-  llvm_ctpop_i32: function(x) {
```

Comment:
1) Basically, when something has not been implemented otherwise, we look for it in this file. Since this isn't implemented by the backend, it ends up taken from here.

2) Maybe there hasn't been a need for them, so we never implemented them. We basically implement what clang emits, which is a small subset of LLVM IR.

4) I would add a test for a bunch of values, in particular 0, all bits set, and a few others. I agree adding all 2**32 is too much.